diff -ruN hyprland-qtutils-0.1.5.og/utils/dialog/CMakeLists.txt hyprland-qtutils-0.1.5/utils/dialog/CMakeLists.txt
--- hyprland-qtutils-0.1.5.og/utils/dialog/CMakeLists.txt	2025-09-12 02:17:25.162703409 +0100
+++ hyprland-qtutils-0.1.5/utils/dialog/CMakeLists.txt	2025-09-12 02:17:48.777018739 +0100
@@ -5,7 +5,7 @@
 set(CMAKE_CXX_STANDARD 23)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-find_package(Qt6 6.5 REQUIRED COMPONENTS Widgets Quick QuickControls2 WaylandClient)
+find_package(Qt6 6.5 REQUIRED COMPONENTS Widgets Quick QuickControls2 WaylandClient WaylandClientPrivate)
 find_package(PkgConfig REQUIRED)
 
 pkg_check_modules(hyprutils REQUIRED IMPORTED_TARGET hyprutils)
diff -ruN hyprland-qtutils-0.1.5.og/utils/donate-screen/CMakeLists.txt hyprland-qtutils-0.1.5/utils/donate-screen/CMakeLists.txt
--- hyprland-qtutils-0.1.5.og/utils/donate-screen/CMakeLists.txt	2025-09-12 02:17:33.262560910 +0100
+++ hyprland-qtutils-0.1.5/utils/donate-screen/CMakeLists.txt	2025-09-12 02:17:52.870832991 +0100
@@ -5,7 +5,7 @@
 set(CMAKE_CXX_STANDARD 23)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-find_package(Qt6 6.5 REQUIRED COMPONENTS Widgets Quick QuickControls2 WaylandClient)
+find_package(Qt6 6.5 REQUIRED COMPONENTS Widgets Quick QuickControls2 WaylandClient WaylandClientPrivate)
 find_package(PkgConfig REQUIRED)
 
 pkg_check_modules(hyprutils REQUIRED IMPORTED_TARGET hyprutils)
diff -ruN hyprland-qtutils-0.1.5.og/utils/update-screen/CMakeLists.txt hyprland-qtutils-0.1.5/utils/update-screen/CMakeLists.txt
--- hyprland-qtutils-0.1.5.og/utils/update-screen/CMakeLists.txt	2025-09-12 02:17:39.778745994 +0100
+++ hyprland-qtutils-0.1.5/utils/update-screen/CMakeLists.txt	2025-09-12 02:17:56.597952730 +0100
@@ -5,7 +5,7 @@
 set(CMAKE_CXX_STANDARD 23)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-find_package(Qt6 6.5 REQUIRED COMPONENTS Widgets Quick QuickControls2 WaylandClient)
+find_package(Qt6 6.5 REQUIRED COMPONENTS Widgets Quick QuickControls2 WaylandClient WaylandClientPrivate)
 find_package(PkgConfig REQUIRED)
 
 pkg_check_modules(hyprutils REQUIRED IMPORTED_TARGET hyprutils)
